### ISSUE TITLE
label-has-for false alarm on version 6.0.2

### ISSUE THREAD

**Issue by anhhnt** [evcohen/eslint-plugin-jsx-a11y#308]
Hello,

I got error `label-has-for` raised for this simple code:

`const App = () =>
    <div>
        <input type="checkbox" id="abc" />
        <label htmlFor="abc" />
    </div>;
`

However, it didn't in version 5.x.x.
Here is the complete setup that produce the issue:
https://github.com/anhhnt/eslint-htmlfor

**Comment by ljharb** [comment]
The label should be wrapping the input as well as being id/for-linked.

**Comment by anhhnt** [comment]
Thanks @ljharb, I wonder why we're forcing `label` to wrap `input`, as the recommendation from W3C still put `label` as sibling of `input`.

**Comment by ljharb** [comment]
Some devices only work with wrapping, some only work with for/ID linking; doing both is required to cover 100% of devices.

You can reconfigure it, if needed, to only require one or the other.

**Comment by anhhnt** [comment]
Thank you very much! I didn't carefully read the documentation of the rule

**Comment by langpavel** [comment]
Huh, I think that this code is OK, but now I'm getting error. For me, dumb, please, can anybody explain?
```js
            <div className={s.formGroup}>
              <label className={s.label} htmlFor="usernameOrEmail">
                Username or email address:
              </label>
              <input
                className={s.input}
                id="usernameOrEmail"
                type="text"
                name="usernameOrEmail"
                autoFocus // eslint-disable-line jsx-a11y/no-autofocus
              />
            </div>
```
What is bad with this?

**Comment by ljharb** [comment]
The input needs to go inside the label, yes.

A PR to improve the error message is always appreciated :-)

**Comment by langpavel** [comment]
@ljharb Thanks for explanation :smile: 
I know how simple PR can be difficult for someone to do. Can you (or someone else) be helpful, provide link please? And of course, what should be inside?
I propose: `'Form label must be wrapper of the control. See #XXX for more details'`

**Comment by anhhnt** [comment]
I can help with the PR ;-)

**Comment by jessebeach** [comment]
> Some devices only work with wrapping, some only work with for/ID linking; doing both is required to cover 100% of devices.

@ljharb have you seen any research on what these devices are and their capabilities? I'm not convinced that wrapping is required in for modern UAs.

**Comment by ljharb** [comment]
I've not seen any research besides a few articles in the mid-to-late 2000s; my experience tells me that all *modern* UAs do indeed work fine with just nesting - but given that specifying an ID isn't that difficult, and that it ensures 100% coverage on all devices, I don't personally find the burden of proof to be very high to convince me to do it.
